executor/diagnostics_test.go

```diff
@@ -166,6 +171,140 @@ func (s *diagnosticsSuite) TestInspectionResult(c *C) {
 	}
 }
 
+func (s *diagnosticsSuite) TestThresholdCheckInspection(c *C) {
+	// mock TiKV http server
+	router := mux.NewRouter()
+	type mockServer struct {
+		address string
+		server  *httptest.Server
+	}
+	server := httptest.NewServer(router)
+	address := strings.TrimPrefix(server.URL, "http://")
+	testServer := &mockServer{
+		address: address,
+		server:  server,
+	}
+	defer func() { testServer.server.Close() }()
+	var mockTiKVConfig = func() (map[string]interface{}, error) {
+		configuration := map[string]interface{}{
+			"raftstore.apply-pool-size":               "2",
+			"raftstore.store-pool-size":               "2",
+			"readpool.coprocessor.normal-concurrency": "4",
+			"readpool.coprocessor.high-concurrency":   "4",
+			"readpool.coprocessor.low-concurrency":    "4",
+			"readpool.storage.high-concurrency":       "4",
+			"readpool.storage.low-concurrency":        "4",
+			"readpool.storage.normal-concurrency":     "4",
+			"server.grpc-concurrency":                 "8",
+			"storage.scheduler-worker-pool-size":      "6",
+		}
+		return configuration, nil
+	}
+	router.Handle("/config", fn.Wrap(mockTiKVConfig))
+
+	fpName := "github.com/pingcap/tidb/executor/mockClusterConfigServerInfo"
+	fpExpr := strings.Join([]string{"tikv", testServer.address, testServer.address}, ",")
+	c.Assert(failpoint.Enable(fpName, fmt.Sprintf(`return("%s")`, fpExpr)), IsNil)
+	defer func() { c.Assert(failpoint.Disable(fpName), IsNil) }()
+
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustQuery("select type, `key`, value from information_schema.cluster_config where type='tikv'").Check(testkit.Rows(
+		"tikv raftstore.apply-pool-size 2",
+		"tikv raftstore.store-pool-size 2",
+		"tikv readpool.coprocessor.high-concurrency 4",
+		"tikv readpool.coprocessor.low-concurrency 4",
+		"tikv readpool.coprocessor.normal-concurrency 4",
+		"tikv readpool.storage.high-concurrency 4",
+		"tikv readpool.storage.low-concurrency 4",
+		"tikv readpool.storage.normal-concurrency 4",
+		"tikv server.grpc-concurrency 8",
+		"tikv storage.scheduler-worker-pool-size 6",
+	))
+
+	// Mock for metric table data.
+	fpName2 := "github.com/pingcap/tidb/executor/mockMetricsTableData"
+	c.Assert(failpoint.Enable(fpName2, "return"), IsNil)
+	defer func() { c.Assert(failpoint.Disable(fpName2), IsNil) }()
+
+	datetime := func(s string) types.Time {
+		t, err := types.ParseTime(tk.Se.GetSessionVars().StmtCtx, s, mysql.TypeDatetime, types.MaxFsp)
+		c.Assert(err, IsNil)
+		return t
+	}
```

Comment:
Extract this function, here is a [same function definition](https://github.com/pingcap/tidb/blob/5d4eb4c806d07382043d795268497071d0f6ca5a/executor/diagnostics_test.go#L176-L180)